agent/xds/routes.go

```diff
@@ -430,6 +425,85 @@ func (s *ResourceGenerator) routesForIngressGateway(cfgSnap *proxycfg.ConfigSnap
 	return result, nil
 }
 
+// routesForAPIGateway returns the xDS API representation of the
+// "routes" in the snapshot.
+func (s *ResourceGenerator) routesForAPIGateway(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
+	var result []proto.Message
+
+	readyUpstreamsList := getReadyUpstreams(cfgSnap)
+
+	for _, readyUpstreams := range readyUpstreamsList {
+		listenerCfg := readyUpstreams.listenerCfg
+		// Do not create any route configuration for TCP listeners
+		if listenerCfg.Protocol != "http" {
```

Comment:
minor but do we have `"http"` defined as a constant somewhere to use rather than just the string?

Reply:
and would `http` also encapsulate if the protocol was `https`? (is `https` even a vaild protocol we could have?)

Reply:
Our validation ensures that the only two values for protocol will ever be "tcp" or "http". As for if its defined as a constant anywhere, as far as I could tell everywhere else in the code is explicitly referring to "tcp" and "http" so I was keeping it consistent, but I'm not really sure why we don't have http and tcp defined as constants so I'm not opposed to adding that in.

Reply:
Never mind, found a constant will update to use it.